File: 238-Product-of-Array-Except-Self.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        n = len(nums)
        zero_count= nums.count(0)
        product = prod(num if num != 0 else 1 for num in nums[:n])
        if zero_count ==1:
            return [product if num ==0 else 0 for num in nums[:n]]
        elif zero_count > 1:
            return [0 for _ in nums]
        else:
            return [product // num for num in nums[:n]]

      


        

File: 238-Product-of-Array-Except-Self.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int zero_count = count(begin(nums),end(nums),0);
        int product =1;
        for(int num : nums) product*= num!= 0 ? num : 1;
        int n = nums.size();
        vector<int> res(n);
        for(int i=0;i<n;i++){
                if(zero_count==1){
                    if(nums[i]==0)res[i]=product;
                    else res[i]=0;
                }else if(zero_count>1){
                    res[i]=0;
                }else{
                    res[i]=product/nums[i];
                }
        } 
        return res;
    }
};